staging/src/k8s.io/dynamic-resource-allocation/resourceclaim/resourceclaim.go

```diff
@@ -78,6 +80,70 @@ func Name(pod *v1.Pod, podClaim *v1.PodResourceClaim) (name *string, mustCheckOw
 	}
 }
 
+// NewNameLookup returns an object which handles determining the name of
+// a ResourceClaim. In contrast to the stand-alone Name it is compatible
+// also with Kubernetes < 1.28.
+//
+// Providing a client is optional. If none is available, then code can pass nil
+// and users can set the DRA_WITH_DETERMINISTIC_RESOURCE_CLAIM_NAMES env
+// variable to an arbitrary non-empty value to use the naming from Kubernetes <
+// 1.28.
+func NewNameLookup(client kubernetes.Interface) *Lookup {
```

Comment:
Keeping this function easy and not doing anything which can fail is intentional because it avoids adding additional error checks in code which uses this.

Reply:
One question: Are there other options that we would want to pass to the contructor (something like `withDeterministicResourceClaimNames` to allow for overriding this behaviour in tests)?